src/main/scala/mesosphere/marathon/core/health/impl/AppHealthCheckActor.scala

```diff
@@ -0,0 +1,89 @@
+package mesosphere.marathon
+package core.health.impl
+
+import akka.actor.{ Actor, Props }
+import akka.event.EventStream
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.event.InstanceHealthChanged
+import mesosphere.marathon.core.health.impl.AppHealthCheckActor.{ AddHealthCheck, ApplicationKey, HealthCheckStatusChanged, RemoveHealthCheck }
+import mesosphere.marathon.core.health.{ Health, HealthCheck }
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.state.{ PathId, Timestamp }
+
+class AppHealthCheckActor(eventBus: EventStream) extends Actor with StrictLogging {
+  private var healthChecks: Map[ApplicationKey, Set[HealthCheck]] = Map.empty
+  private var healthCheckStates: Map[(ApplicationKey, Instance.Id), (Map[HealthCheck, Option[Health]], Option[Boolean])] =
```

Comment:
OPT: remove the global status, recompute each time => simplify data structure and be less error prone